internal/directory/auth0/auth0.go

```diff
@@ -253,20 +258,23 @@ func parseServiceAccountFromOptions(clientID, clientSecret string) (*ServiceAcco
 }
 
 func parseServiceAccountFromString(rawServiceAccount string) (*ServiceAccount, error) {
-	bs, err := base64.StdEncoding.DecodeString(rawServiceAccount)
-	if err != nil {
-		return nil, fmt.Errorf("auth0: could not decode base64: %w", err)
-	}
-
 	var serviceAccount ServiceAccount
-	if err := json.Unmarshal(bs, &serviceAccount); err != nil {
+	if err := encoding.DecodeBase64OrJSON(rawServiceAccount, &serviceAccount); err != nil {
 		return nil, fmt.Errorf("auth0: could not unmarshal json: %w", err)
 	}
 
 	if serviceAccount.ClientID == "" {
 		return nil, errors.New("auth0: client_id is required")
 	}
 
+	// for backwards compatibility we support secret and client_secret
+	if serviceAccount.Secret == "" {
+		serviceAccount.Secret = serviceAccount.ClientSecret
+	}
+	if serviceAccount.ClientSecret == "" {
+		serviceAccount.ClientSecret = serviceAccount.Secret
```

Comment:
It's a little confusing both ends up being set. I think it'd leave less room for future ambiguity if the backwards compatibility only extends to reading the data, but the code always has to use one.